tools/tools.go

```diff
@@ -370,6 +370,15 @@ func EnvmanRun(envstorePth, workDirPth string, cmdArgs []string, timeout time.Du
 	cmd.AppendEnv("PWD=" + workDirPth)
 
 	err := cmd.Start()
+
+	// flush the writer anyway if the process is finished
+	if configs.IsSecretFiltering {
```

Comment:
for me it seems the last write will do the flush if there is any leftover chunk, am i missing something?

Reply:
When the command closes, the timers set will be destroyed so it won't be able to call flush, or maybe it will be flushed in the wrong time.

This way we can make sure that if the command closes, the remaining relevant info will be written there.